src/cugw_qpt.cu

```diff
@@ -349,7 +349,11 @@ namespace green::gpu {
         throw std::runtime_error("failure allocating Gk_tsij on host");
       if (cudaMallocHost(&Gk_smtij_buffer_, ns_ * ntnao2_ * sizeof(cxx_complex)) != cudaSuccess)
         throw std::runtime_error("failure allocating Gk_tsij on host");
-      Sigmak_stij_buffer_ = Gk_smtij_buffer_;
+      // ! GH:  I think this will interfere with our cudaMemcpyAsync. Should we simply allocate a different array for Sigmak_stij_buffer_?
+      // !      The more I think, this here is the real reason why we had to use cudaMemcpy and not the asynchronous version.
+      // ! <previously> Sigmak_stij_buffer_ = Gk_smtij_buffer_;
+      if (cudaMallocHost(&Sigmak_stij_buffer_, ns_ * ntnao2_ * sizeof(cxx_complex)) != cudaSuccess)
+        throw std::runtime_error("failure allocating Gk_tsij on host");
```

Comment:
sure we can allocate another buffer on the host. That's totally fine, no reason to preserve memory here.